modules/codegen-plugin/src/smithy4s/codegen/CodegenPlugin.scala

```diff
@@ -133,44 +139,33 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
         m.root
       }
     val transforms = (conf / smithy4sModelTransformers).value
-    val s = streams.value
-
-    val cached =
-      Tracked.inputChanged[CacheKey, Seq[File]](
-        s.cacheStoreFactory.make("input")
-      ) {
-        untupled {
-          Tracked
-            .lastOutput[(Boolean, CacheKey), Seq[File]](
-              s.cacheStoreFactory.make("output")
-            ) { case ((changed, _), outputs) =>
-              if (changed || outputs.isEmpty) {
-                val filePaths = inputFiles.map(_.getAbsolutePath())
-                val codegenArgs = CodegenArgs(
-                  filePaths.map(os.Path(_)).toList,
-                  output = os.Path(outputPath),
-                  openapiOutput = os.Path(openApiOutputPath),
-                  skipScala = false,
-                  skipOpenapi = false,
-                  allowedNS = allowedNamespaces,
-                  excludedNS = excludedNamespaces,
-                  repositories = res,
-                  dependencies = dependencies,
-                  transforms
-                )
-                val resPaths = smithy4s.codegen.Codegen
-                  .processSpecs(codegenArgs)
-                  .toList
-                resPaths.map(path => new File(path.toString))
-              } else {
-                outputs.getOrElse(Seq.empty)
-              }
-            }
-        }
-      }
 
-    cached(
-      (FilesInfo(inputFiles.map(FileInfo.hash(_)).toSet), dependencies)
-    )
+    val fileChangesReport = (conf / smithy4sCodegen).inputFileChanges
+
+    if (fileChangesReport.hasChanges) {
+
+      import fileChangesReport.*
+
+      val filePaths = modified ++ created ++ unmodified
```

Comment:
The general idea is simple - if there were _any_ changes, we just reconstruct the "live" set of smithy files (which means unmodified + modified - deleted)

If there were no changes, we need to return the list of output files still - otherwise SBT won't pick them up for future compilation.

Reply:
Would you mind inlining that as a comment in the code ? That's useful information to have when we read the files

Reply:
that's amazing I did not know sbt had `inputFileChanges` along with `fileInputs` and `fileOutputs`